lib/pages_desktop_specific/my_stack_view/wallet_view/sub_widgets/more_features/more_features_dialog.dart

```diff
@@ -136,6 +151,58 @@ class _MoreFeaturesDialogState extends ConsumerState<MoreFeaturesDialog> {
               iconAsset: Assets.svg.cashFusion,
               onPressed: () => widget.onFusionPressed?.call(),
             ),
+          if (wallet.info.coin is Firo)
+            Padding(
+              padding: const EdgeInsets.symmetric(
+                vertical: 6,
+                horizontal: 32,
+              ),
+              child: RoundedContainer(
+                color: Colors.transparent,
+                borderColor: Theme.of(context)
+                    .extension<StackColors>()!
+                    .textFieldDefaultBG,
+                child: Row(
+                  children: [
+                    SizedBox(width: 3),
+                    SizedBox(
+                      height: 20,
+                      width: 40,
+                      child: DraggableSwitchButton(
+                        isOn: enableLelantusScanning ?? false,
+                        onValueChanged: (newValue) {
+                          // Toggle enableLelantusScanning in wallet info.
+                          wallet.info.updateOtherData(newEntries: {
+                            WalletInfoKeys.enableLelantusScanning:
+                                !(enableLelantusScanning ?? false)
+                          }, isar: ref.read(mainDBProvider).isar).then((value) {
+                            // Should setState be used here?
```

Comment:
Yes probably. I would also use a simple bool value mutex to lock this function from being called again until it has completed.

Reply:
Resolved by aad422d92e808384068273e7f8ef0627161d711c